src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -764,7 +774,11 @@ class Simplex_tree {
   Simplex_handle find_vertex(Vertex_handle v) {
     if (Options::contiguous_vertices) {
```

Comment:
I think **all** tests of Options::contiguous_vertices should also test `!Options::simplex_handle_strong_validity` (or we could have a static_assert forbidding to have both, though that doesn't sound as good). Then you shouldn't need the 2-argument version of `std::next` anywhere.

Reply:
I replaced them all. I also converted them to constexpr.
It is probably better to allow them both in case we find some other optimization where both are possible?